app/scripts/modules/core/src/managed/ManagedResourceObject.tsx

```diff
@@ -21,23 +22,50 @@ const kindIconMap: { [kind: string]: IconNames } = {
   'application-load-balancer': 'loadBalancer',
 };
 
-function getIconTypeFromKind(kind: string) {
-  return kindIconMap[getKindName(kind)] ?? 'placeholder';
-}
+const getIconTypeFromKind = (kind: string) => kindIconMap[getKindName(kind)] ?? 'placeholder';
+
+const getResourceName = ({ moniker: { app, stack, detail } }: IManagedResourceSummary) =>
+  [app, stack, detail].filter(Boolean).join('-');
 
-export const ManagedResourceObject = ({
-  resource: {
+const getResourceRoutingInfo = (
+  resource: IManagedResourceSummary,
+): { state: string; params: { [key: string]: string } } | null => {
```

Comment:
we're not using the `strictNullChecks` flag on TS, so the `| null` union is already done for you (not that I'm saying that's a good thing...)